### PR TITLE
Add GCP IAM restrictions checks

### PR DESCRIPTION
**Pull request by arielkru**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Inspired by [Compromise any GCP Org Via Cloud API Lateral Movement and Privilege Escalation: Blackhat/Defcon 2020](https://www.youtube.com/watch?v=Ml09R38jpok)

### CODE REVIEW

File: checkov/terraform/checks/resource/gcp/AbsGoogleImpersonationRoles.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "roles/dataproc.serviceAgent",
+]
+
+
+class AbsGoogleImpersonationRoles(BaseResourceCheck):
+    def __init__(self, name, id, categories, supported_resources):
+        super().__init__(name, id, categories, supported_resources)
+
+    def scan_resource_conf(self, conf):
+        if 'role' in conf and conf['role'][0] in IMPERSONATION_ROLES:
```
Comment (nimrodkor) [inline checkov/terraform/checks/resource/gcp/AbsGoogleImpersonationRoles.py:30]: @arielkru Why not iterate through the roles? Why take only the first role at index 0?

**Review by nimrodkor — COMMENTED** [review]
Awesome work @arielkru !

File: checkov/terraform/checks/resource/gcp/AbsGoogleImpersonationRoles.py (hunk 1 above)
Comment (arielkru) [reply checkov/terraform/checks/resource/gcp/AbsGoogleImpersonationRoles.py:30]: @nimrodkor Can the role value be an array? I know it gets only a single string

**Review by nimrodkor — APPROVED** [review]
@arielkru You had me at `ternary operator`

**Pull request merged**